x-pack/plugins/security_solution/public/flyout/right/components/correlations_overview.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { useExpandableFlyoutContext } from '@kbn/expandable-flyout';
+import { useFetchRelatedAlertsBySameSourceEvent } from '../hooks/use_fetch_related_alerts_by_same_source_event';
+import { useShowRelatedAlertsBySameSourceEvent } from '../hooks/use_show_related_alerts_by_same_source_event';
+import { useFetchRelatedAlertsBySession } from '../hooks/use_fetch_related_alerts_by_session';
+import { useShowRelatedAlertsBySession } from '../hooks/use_show_related_alerts_by_session';
+import { useFetchRelatedAlertsByAncestry } from '../hooks/use_fetch_related_alerts_by_ancestry';
+import { useShowRelatedAlertsByAncestry } from '../hooks/use_show_related_alerts_by_ancestry';
+import { useShowRelatedCases } from '../hooks/use_show_related_cases';
+import { useFetchRelatedCases } from '../hooks/use_fetch_related_cases';
+import { INSIGHTS_CORRELATIONS_TEST_ID } from './test_ids';
+import { InsightsSubSection } from './insights_subsection';
+import type { InsightsSummaryPanelData } from './insights_summary_panel';
+import { InsightsSummaryPanel } from './insights_summary_panel';
+import { useRightPanelContext } from '../context';
+import {
+  CORRELATIONS_ANCESTRY_ALERT,
+  CORRELATIONS_ANCESTRY_ALERTS,
+  CORRELATIONS_RELATED_CASE,
+  CORRELATIONS_RELATED_CASES,
+  CORRELATIONS_SAME_SESSION_ALERT,
+  CORRELATIONS_SAME_SESSION_ALERTS,
+  CORRELATIONS_SAME_SOURCE_EVENT_ALERT,
+  CORRELATIONS_SAME_SOURCE_EVENT_ALERTS,
+  CORRELATIONS_TEXT,
+  CORRELATIONS_TITLE,
+  VIEW_ALL,
+} from './translations';
+import { LeftPanelKey, LeftPanelInsightsTabPath } from '../../left';
+
+/**
+ * Correlations section under Insights section, overview tab.
+ * The component fetches the necessary data, then pass it down to the InsightsSubSection component for loading and error state,
+ * and the SummaryPanel component for data rendering.
+ */
+export const CorrelationsOverview: React.FC = () => {
+  const { eventId, indexName, browserFields, dataFormattedForFieldBrowser, scopeId } =
+    useRightPanelContext();
+  const { openLeftPanel } = useExpandableFlyoutContext();
+
+  const goToCorrelationsTab = useCallback(() => {
+    openLeftPanel({
+      id: LeftPanelKey,
+      path: LeftPanelInsightsTabPath,
+      params: {
+        id: eventId,
+        indexName,
+      },
+    });
+  }, [eventId, openLeftPanel, indexName]);
+
+  const data: InsightsSummaryPanelData[] = [];
```

Comment:
I don't think a mutable operation like this is wise to do in the render of a component, consider a custom hook at this top level that uses all of the newly added custom hooks, and returns the data/state needed for each of the components here, along with a global error state.

Reply:
extracted to new hook, sorry for the oversight!